### PR TITLE
Issue 44110 - Support for displaying run object inputs and object outputs

### PR DESCRIPTION
**Pull request by ankurjuneja**
#### Rationale
This PR provides support to register run inputs and run outputs views with experiment service and display those views in run details view.

#### Related Pull Requests
* https://github.com/LabKey/provenance/pull/82

#### Changes
* View registration support

### CODE REVIEW

File: experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java [hunk 1]
```diff
@@ -7710,6 +7714,30 @@ public boolean useUXDomainDesigner()
...
+    @Override
+    public void registerRunOutputsViewProvider(Set<QueryForm> queryForms)
+    {
+        _runOutputsQueryForms.addAll(queryForms);
+    }
+
+    @Override
+    public Set<QueryForm> getRunInputsQueries()
+    {
+        return _runInputsQueryForms;
```
Comment (labkey-jeckels) [inline experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java]: Please return an immutable wrapper over these collections via Collections.unmodifiableSet()

File: api/src/org/labkey/api/exp/api/ExperimentService.java [hunk 2]
```diff
@@ -792,6 +793,14 @@ ExpRun derive(Map<ExpMaterial, String> inputMaterials, Map<ExpData, String> inpu
 
     SQLFragment generateExperimentTreeSQLLsidSeeds(List<String> lsids, ExpLineageOptions options);
 
+    Set<QueryForm> getRunInputsQueries();
+
+    Set<QueryForm> getRunOutputsQueries();
+
+    void registerRunInputsViewProvider(Set<QueryForm> queryForms);
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/exp/api/ExperimentService.java]: A comment here about where this is used and the expectation that the referenced query has a RunId column would be helpful.

**Review by labkey-jeckels — APPROVED** [review]
Some recommendations but no need for additional review.

File: api/src/org/labkey/api/exp/api/ExperimentService.java (hunk 2 above)
Comment (ankurjuneja) [reply api/src/org/labkey/api/exp/api/ExperimentService.java]: done

File: experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java (hunk 1 above)
Comment (ankurjuneja) [reply experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java]: done

**Pull request merged**